src/hooks/searchParams/useGlobalSearchParams.ts

```diff
@@ -0,0 +1,50 @@
+import { useCallback, useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+export enum globalSearchParams {
+    connectorId = 'connectorId',
+    liveSpecId = 'liveSpecId',
+    lastPubId = 'lastPubId',
+}
+
+function useGlobalSearchParams(key: globalSearchParams): string;
+function useGlobalSearchParams(
+    key: globalSearchParams,
+    fetchAll: true
+): string[];
+function useGlobalSearchParams(
+    key: globalSearchParams | Array<globalSearchParams>
+): string[];
+function useGlobalSearchParams(
+    key: globalSearchParams | Array<globalSearchParams>,
+    fetchAll: true
+): string[][];
+function useGlobalSearchParams(
+    key: globalSearchParams | Array<globalSearchParams>,
+    fetchAll?: boolean
+) {
+    const [searchParams] = useSearchParams();
+
+    const getValue = useCallback(
+        (keyVal: string) => {
```

Comment:
So I thought about that and decided against it last night. Mainly because initially the `getAll` call did some more things before returning. Since I have a feeling those are going to get added back in I left the larger if to allow more flexibility .